WordPress/Classes/ViewRelated/NUX/Signin2FAViewController.swift

```diff
@@ -0,0 +1,248 @@
+
+        configureSendCodeButtonText()
+        configureStatusLabel("")
+        configureSubmitButton(false)
+    }
+
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+
+        configureViewForEditingIfNeeded()
+    }
+
+
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+
+        registerForKeyboardEvents(#selector(SigninEmailViewController.handleKeyboardWillShow(_:)),
+                                  keyboardWillHideAction: #selector(SigninEmailViewController.handleKeyboardWillHide(_:)))
+
+    }
+
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        unregisterForKeyboardEvents()
+
+        // Multifactor codes are time sensitive, so clear the stored code if the
+        // user dismisses the view. They'll need to reentered it upon return. 
+        loginFields.multifactorCode = ""
+        verificationCodeField.text = ""
+    }
+
+
+    // MARK: Configuration Methods
+
+
+    /// Configures the appearance of the button to request a 2fa code be sent via SMS.
+    ///
+    func configureSendCodeButtonText() {
+        // Text: Verification Code SMS
+        let string = NSLocalizedString("Enter the code on your authenticator app or <u>send the code via text message</u>.",
+                                       comment: "Message displayed when a verification code is needed")
+        let options = [
+            NSDocumentTypeDocumentAttribute: NSHTMLTextDocumentType
+        ]
+
+        let styledString = "<style>body {font-family: sans-serif; font-size:14px; color: #ffffff; text-align:center;}</style>" + string
+
+        guard let data = styledString.dataUsingEncoding(NSUTF8StringEncoding),
+            attributedCode = try? NSMutableAttributedString(data: data, options: options, documentAttributes: nil),
+            attributedCodeHighlighted = attributedCode.mutableCopy() as? NSMutableAttributedString
+            else {
+                return
+        }
+
+        attributedCodeHighlighted.applyForegroundColor(WPNUXUtility.confirmationLabelColor())
+
+        if let titleLabel = sendCodeButton.titleLabel  {
+            titleLabel.lineBreakMode = .ByWordWrapping
+            titleLabel.textAlignment = .Center
+            titleLabel.numberOfLines = 3
+        }
+
+        sendCodeButton.setAttributedTitle(attributedCode, forState: .Normal)
+        sendCodeButton.setAttributedTitle(attributedCodeHighlighted, forState: .Highlighted)
+    }
+
+
+    /// Displays the specified text in the status label.
+    ///
+    /// - Parameters:
+    ///     - message: The text to display in the label.
+    ///
+    func configureStatusLabel(message: String) {
+        statusLabel.text = message
+
+        sendCodeButton.hidden = !message.isEmpty
+    }
+
+
+    /// Configures the appearance and state of the submit button.
+    ///
+    func configureSubmitButton(animating: Bool) {
```

Comment:
I'd make `animating` a named parameter (also in the other VCs that have this). When you see something like `configureSubmitButton(false)` it's really hard to know what the parameter means. `configureSubmitButton(animating: false)` is much more obvious.